### ISSUE TITLE
Missing MatterJS TypeScript type definitions for Phaser 3

### ISSUE THREAD

**Issue by colorcube** [photonstorm/phaser#4199]
I tried Phaser 3.15.1 with TypeScript with definitions from here: https://github.com/photonstorm/phaser3-docs/tree/master/typescript

One missing piece is the type definition for MatterJS which is not existent except some empty classes.

There are MatterJS type definitions: https://www.npmjs.com/package/@types/matter-js

Unfortunately they don't work together. In phaser.d.ts the namespace MatterJS is used in /@types/matter-js it is Matter. Just renaming MatterJS to Matter doesn't work. I guess MatterJS is a wrapper anyway.

Long story short: it would be nice to have TypeScript type definitions for MatterJS in Phaser 3

**Comment by glebkudr** [comment]
The same issue for me...

**Comment by pierpo** [comment]
Same issue here 😞
It makes it hard to work with Matter.

Anyone planning on adding the types?

**Comment by matix931** [comment]
I just coppied /@types/matter-js namespace definition in to the proper place: Phaser.Physics.Matter.Matter and it is working perfectly!

Example usage:

```typescript
export class Tower extends GameObjects.Container { }

this.towers = this.add.group({
   classType: Tower as any,
   runChildUpdate: true,
})

let tower = this.towers.create(50, 50) as Tower
// this.matter.add.gameObject adds some additional methods for MatterJS and Phaser compatibility
// those methods are stored in interfaces at Phaser.Physics.Matter.Components.*
const towerMatter = this.matter.add.gameObject(tower, null) as any as Phaser.Physics.Matter.Components.SetBody
towerMatter.setExistingBody(Phaser.Physics.Matter.Matter.Bodies.rectangle(50,50,32,32))
```

Ready to go phaser.d.ts:
[phaser.d.ts.zip](https://github.com/photonstorm/phaser/files/2979293/phaser.d.ts.zip)
*github is not allowing *.ts files

**Comment by ipelekhan** [comment]
@matix931 could you please explain your fix a little bit more? I can't find any `Phase.Physics.Matter.Matter` lines in your `phaser.d.ts.zip` file.

**Comment by steve-salmond** [comment]
@matix931 Thanks very much for posting this workaround - most helpful!

**Comment by photonstorm** [comment]
Thank you for submitting this issue. We have fixed this and the fix has been pushed to the `master` branch. It will be part of the next release. If you get time to build and test it for yourself we would appreciate that.

**Comment by robert-everett-brandt** [comment]
@matix931 - When you say "I just coppied /@types/matter-js namespace definition in to the proper place" what exactly did you mean?  Moving the @types/matter-js folder to a new location?  Or actually copying it's contents to a new spot?

Thanks in advance.

**Comment by photonstorm** [comment]
This isn't needed any longer, Phaser 3 has its own Matter TypeScript defs, specially created just for its implementation of Matter that comes bundled in the types folder.

**Comment by mattmccullo** [comment]
I'm having this same issue now introducing this into my Angular project: `node_modules/phaser/types/phaser.d.ts(69953,46): error TS2503: Cannot find namespace 'MatterJS'.`

I've installed phaser and tried a few times, tried messing around with typing too.

**Comment by photonstorm** [comment]
This issue has been mentioned on **Phaser**. There might be relevant details there:

https://phaser.discourse.group/t/matter-and-typescript/6994/3

**Comment by herrero-tranquilo** [comment]
I have defined in d.ts like this:
```typescript

/// <reference lib="matter" />
/// <reference lib="phaser" />
declare module "phaser" {
  namespace Physics.Matter.Matter {
    export class Body extends MatterJS.Body {
      static create(
        options: MatterJS.IChamferableBodyDefinition
      ): MatterJS.BodyType;
    }
    export class Bodies extends MatterJS.Bodies {}
  }
}

const { Body, Bodies } = Phaser.Physics.Matter.Matter;

```

**Comment by jl91** [comment]
any update about that issue ? i'm using phaser 3 with angular 13 and phaser-matter plugin
habing the same problem

**Comment by diandian18** [comment]
Still Waiting for the solution

**Comment by photonstorm** [comment]
Because commenting on dead issues is always the way to find those.
